tests/netlify-deploy.ts

```diff
@@ -169,6 +174,32 @@ export class NextDeployInstance extends NextInstance {
     ).trim()
 
     require('console').log(`Got buildId: ${this._buildId}`)
+    require('console').log(`Setup time: ${(Date.now() - setupStartTime) / 1000.0}s`)
+  }
+
+  public async destroy(): Promise<void> {
+    if (this._shouldDeleteDeploy) {
+      require('console').log(`Deleting project with deploy_id ${this._deployId}`)
+
+      const deleteResponse = await execa('npx', [
+        'ntl',
+        'api',
+        'deleteDeploy',
+        '--data',
+        `{ "deploy_id": "${this._deployId}" }`,
+      ])
+
+      if (deleteResponse.exitCode !== 0) {
+        require('console').error(
+          `Failed to delete deploy ${deleteResponse.stdout} ${deleteResponse.stderr} (${deleteResponse.exitCode})`,
+        )
+      } else {
+        require('console').log(`Successfully deleted deploy with deploy_id ${this._deployId}`)
+        this._shouldDeleteDeploy = false
```

Comment:
learning - why do we need to set this to false in the end? will it keep attempting and looping if we don't?

Reply:
It should not, this is just in case - I was looking at base class we extend - it does have some checks that look if something was deleted / destroyed already - there they throw:

https://github.com/vercel/next.js/blob/2e28c965279de90ce4bfca673196c27dd6117027/test/lib/next-modes/base.ts#L425-L428

but I didn't want to play with this as I don't fully know the contract that Vercel tests have with those modules - so this just felt safe way to just avoid wasting time on calling API to delete deploy that was already deleted in case this is called multiple times (`deleteDeploy` api call is not immediate - it takes couple seconds to finish so just no point in calling it again)